src/transaction/declare.rs

```diff
@@ -433,575 +570,421 @@ impl Declare {
-        let mut state_reader = InMemoryStateReader::default();
-        state_reader
-            .address_to_class_hash_mut()
-            .insert(sender_address.clone(), class_hash);
-        state_reader
-            .address_to_nonce_mut()
-            .insert(sender_address, Felt252::ZERO);
-
-        let mut state = CachedState::new(Arc::new(state_reader), Arc::new(contract_class_cache));
+        let casm_class =
+            CasmContractClass::from_contract_class(sierra_contract_class.clone(), true).unwrap();
+        let casm_class_hash = compute_casm_class_hash(&casm_class).unwrap();
 
-        //* ---------------------------------------
-        //*    Test declare with previous data
-        //* ---------------------------------------
+        // create internal declare v2
 
-        let fib_contract_class =
-            ContractClass::from_path("starknet_programs/fibonacci.json").unwrap();
-
-        let chain_id = StarknetChainId::TestNet.to_felt();
-
-        // Declare same class twice
-        let internal_declare = Declare::new(
-            fib_contract_class.clone(),
-            chain_id,
-            Address(Felt252::ONE),
-            0,
-            1.into(),
-            Vec::new(),
+        let internal_declare = Declare::new_with_tx_hash(
+            &sierra_contract_class,
+            Some(casm_class),
+            casm_class_hash,
+            sender_address,
+            Default::default(),
+            version,
+            [1.into()].to_vec(),
             Felt252::ZERO,
-        )
-        .unwrap();
-
-        let second_internal_declare = Declare::new(
-            fib_contract_class,
-            chain_id,
-            Address(Felt252::ONE),
-            0,
-            1.into(),
-            Vec::new(),
             Felt252::ONE,
         )
         .unwrap();
 
-        internal_declare
-            .execute(
-                &mut state,
-                &BlockContext::default(),
-                #[cfg(feature = "cairo-native")]
-                None,
-            )
-            .unwrap();
+        // crate state to store casm contract class
+        let casm_contract_class_cache = PermanentContractClassCache::default();
+        let state_reader = Arc::new(InMemoryStateReader::default());
+        let mut state = CachedState::new(state_reader, Arc::new(casm_contract_class_cache));
 
-        assert!(state.get_contract_class(&class_hash).is_ok());
+        // call compile and store
+        assert!(internal_declare
+            .compile_and_store_casm_class(&mut state)
+            .is_ok());
 
-        second_internal_declare
-            .execute(
-                &mut state,
-                &BlockContext::default(),
-                #[cfg(feature = "cairo-native")]
-                None,
-            )
-            .unwrap();
+        // test we  can retreive the data
+        let expected_casm_class = CasmContractClass::from_contract_class(
+            internal_declare.sierra_contract_class.unwrap(),
```

Comment:
the `internal_declare` should be renamed to `declare_tx` or something like that to be consistent with the rest of the changes.